src/Microsoft.ML.Api/TypedCursor.cs

```diff
@@ -103,11 +103,11 @@ private TypedCursorable(IHostEnvironment env, IDataView data, bool ignoreMissing
                     throw _host.Except("Column '{0}' not found in the data view", col.ColumnName);
                 }
                 var realColType = _data.Schema.GetColumnType(colIndex);
-                if (!IsCompatibleType(realColType, col.FieldInfo))
+                if (!IsCompatibleType(realColType, col.MemberInfo))
                 {
                     throw _host.Except(
-                        "Can't bind the IDataView column '{0}' of type '{1}' to field '{2}' of type '{3}'.",
-                        col.ColumnName, realColType, col.FieldInfo.Name, col.FieldInfo.FieldType.FullName);
+                        "Can't bind the IDataView column '{0}' of type '{1}' to field or property '{2}' of type '{3}'.",
+                        col.ColumnName, realColType, col.MemberInfo.Name, col.OutputType.FullName);
```

Comment:
>OutputType [](start = 78, length = 10)

This look something suspicious. In case of IsComputed = true you will return something else rather than FieldType or PropertyType.